src/internal/base-component/metrics/metrics.ts

```diff
@@ -2,45 +2,45 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { CLogClient, PanoramaClient, MetricsV2EventItem } from './log-clients';
-import { buildMetricDetail, buildMetricName, jsonStringify } from './formatters';
-import { ComponentConfiguration, MetricsLogItem } from './interfaces';
+import { buildComponentMetricDetail, buildMetricDetail, getMajorVersion } from './formatters';
+import { ComponentConfiguration, ComponentMetricDetail, PackageSettings } from './interfaces';
 
 const oneTimeMetrics = new Set<string>();
 
-// In case we need to override the theme for VR.
-let theme = '';
-function setTheme(newTheme: string) {
-  theme = newTheme;
-}
-
 export class Metrics {
-  readonly source: string;
-  readonly packageVersion: string;
-
-  private clog = new CLogClient();
-  private panorama = new PanoramaClient();
-
-  constructor(source: string, packageVersion: string) {
-    this.source = source;
-    this.packageVersion = packageVersion;
+  private readonly context: PackageSettings;
+  private readonly clog = new CLogClient();
+  private readonly panorama = new PanoramaClient();
+
+  constructor(packageSource: PackageSettings);
+  constructor(packageSource: string, packageVersion: string);
+  constructor(...args: [PackageSettings] | [string, string]) {
```

Comment:
Keeping dual signature until this one usage is migrated: https://github.com/cloudscape-design/components/blob/3bf581977a630226c8f574d5fa83a8e9c52b2a1c/src/internal/metrics.ts#L7

The rest of the code uses `useComponentMetrics` hook